apiserver/facades/client/modelgeneration/modelgeneration.go

```diff
@@ -20,33 +20,51 @@ import (
 
 var logger = loggo.GetLogger("juju.apiserver.modelgeneration")
 
-// API implements the ModelGenerationAPI interface and is the concrete implementation
-// of the API endpoint.
+// API is the concrete implementation of the API endpoint.
 type API struct {
 	check             *common.BlockChecker
 	authorizer        facade.Authorizer
 	apiUser           names.UserTag
 	isControllerAdmin bool
 	st                State
 	model             Model
+	modelCache        ModelCache
 }
 
-// NewModelGenerationFacade provides the signature required for facade registration.
-func NewModelGenerationFacade(ctx facade.Context) (*API, error) {
+type APIV1 struct {
+	*API
+}
+
+// NewModelGenerationFacadeV2 provides the signature required for facade registration.
+func NewModelGenerationFacadeV2(ctx facade.Context) (*API, error) {
 	authorizer := ctx.Auth()
 	st := &stateShim{State: ctx.State()}
 	m, err := st.Model()
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	return NewModelGenerationAPI(st, authorizer, m)
+	mc, err := ctx.Controller().Model(st.ModelUUID())
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return NewModelGenerationAPI(st, authorizer, m, &modelCacheShim{Model: mc})
```

Comment:
I guess we will be shimming out the `Branch` method to return a wrapper later, despite not needing this shim now.